### DISCUSSION TITLE
Can a template include be used for recursion?

### DISCUSSION THREAD

**Discussion by h3ndrik** [pallets/jinja#2055] [Q&A] [answered]
Another way of phrasing it: Can a template *{% include ... %}* itself?

It doesn't seem like this is documented. It works, though. But I'm not sure if I'm supposed to exploit undocumented behaviour, hence my question.

I'm trying to render a comment tree, including replies to replies. I know there are macros and recursion in for-loops available. But this way I can simplify the code a bit more. Especially compared to macros.

**Comment by davidism** [comment, marked as answer]
Should work fine, did you try it and encounter an issue?

**Reply by h3ndrik** [reply]
Thanks, no issues. Seems to work fine. I just wanted to ask, because I couldn't find any information and the documentation says: "The include tag renders **another** template and [...]". And I suppose technically, "another" doesn't mean itself. I just didn't want to submit code that uses undocumented features to other people's projects... But I'm fine with that answer. And everything seems to work fine. Thanks!
